### ISSUE TITLE
Error during Foxy ROS Build : OSError: Path "." is neither a directory containing a "package.xml" file nor a file

### ISSUE THREAD

**Issue by willofdiamond** [ros/kdl_parser#49]
I am trying to install foxy Ros2 on to ubuntu 18.04.05 using the folling website`https://index.ros.org/doc/ros2/Installation/Foxy/Linux-Development-Setup/` I am getting the following error with  kdl_parser_py during the building process using `colcon build --symlink-install --merge-install` . Not sure what is causing the input of wrong directory and how to fix it?

```
Starting >>> kdl_parser_py
Finished <<< examples_rclpy_minimal_service [4.74s]                                                                                                                         
Starting >>> laser_geometry
Finished <<< examples_rclpy_minimal_subscriber [5.04s]                                                                                                     
Starting >>> launch_testing                                                                                      
--- stderr: kdl_parser_py
Traceback (most recent call last):
  File "setup.py", line 8, in <module>
    package_dir={'': ''}
  File "/usr/lib/python3/dist-packages/catkin_pkg/python_setup.py", line 74, in generate_distutils_setup
    package = parse_package(package_xml_path)
  File "/usr/lib/python3/dist-packages/catkin_pkg/package.py", line 569, in parse_package
    xml, filename = _get_package_xml(path)
  File "/usr/lib/python3/dist-packages/catkin_pkg/package.py", line 502, in _get_package_xml
    raise IOError('Path "%s" is neither a directory containing a "%s" file nor a file' % (path, PACKAGE_MANIFEST_FILENAME))
OSError: Path "." is neither a directory containing a "package.xml" file nor a file
---
Failed   <<< kdl_parser_py [2.69s, exited with code 1]
Finished <<< examples_tf2_py [5.05s]
Aborted  <<< launch_testing [0.07s]
Aborted  <<< laser_geometry [3.14s]                             

Summary: 32 packages finished [2min 15s]
  1 package failed: kdl_parser_py
  2 packages aborted: laser_geometry launch_testing
  2 packages had stderr output: kdl_parser_py laser_geometry
  44 packages not processed

```

**Comment by clalancette** [comment]
So the really weird thing about this is that `kdl_parser_py` is AMENT_IGNORE in Foxy.  That means that colcon should never even try to compile it, never mind run into an error with it.  Can you give the exact steps you used to get to this point?

**Comment by willofdiamond** [comment]
I followed the steps listed in the setup guide `https://index.ros.org/doc/ros2/Installation/Foxy/Linux-Development-Setup/` 

```
locale 
sudo apt update && sudo apt install locales
sudo locale-gen en_US en_US.UTF-8
sudo update-locale LC_ALL=en_US.UTF-8 LANG=en_US.UTF-8
export LANG=en_US.UTF-8

locale  # verify settings
```

```
sudo apt update && sudo apt install curl gnupg2 lsb-release
curl -s https://raw.githubusercontent.com/ros/rosdistro/master/ros.asc | sudo apt-key add -
```



```
sudo apt update && sudo apt install -y \
  build-essential \
  cmake \
  git \
  libbullet-dev \
  python3-colcon-common-extensions \
  python3-flake8 \
  python3-pip \
  python3-pytest-cov \
  python3-rosdep \
  python3-setuptools \
  python3-vcstool \
  
wget
# install some pip packages needed for testing
python3 -m pip install -U \
  argcomplete \
  flake8-blind-except \
  flake8-builtins \
  flake8-class-newline \
  flake8-comprehensions \
  flake8-deprecated \
  flake8-docstrings \
  flake8-import-order \
  flake8-quotes \
  pytest-repeat \
  pytest-rerunfailures \
  pytest
# install Fast-RTPS dependencies
sudo apt install --no-install-recommends -y \
  libasio-dev \
  libtinyxml2-dev
# install Cyclone DDS dependencies
sudo apt install --no-install-recommends -y \
  libcunit1-dev

```

```
mkdir -p ~/ros2_foxy/src
cd ~/ros2_foxy
wget https://raw.githubusercontent.com/ros2/ros2/foxy/ros2.repos
vcs import src < ros2.repos
```
```
rosdep update
rosdep install --from-paths src --ignore-src --rosdistro foxy -y --skip-keys "console_bridge fastcdr fastrtps rti-connext-dds-5.3.1 urdfdom_headers"

```

```
cd ~/ros2_foxy/
colcon build --symlink-install
```

**Comment by clalancette** [comment]
So I tried to reproduce this problem, and I found out that Foxy does indeed have a problem compiling on Ubuntu 18.04.  I've submitted a patch here: https://github.com/ros2/libyaml_vendor/pull/32 .

However, with that fixed locally, I couldn't reproduce your problem.  The only thing I can think is that somehow the AMENT_IGNORE file from src/ros/kdl_parser/kdl_parser_py is missing, because otherwise `colcon` should not have been in that directory at all.
